layers/+lang/typescript/packages.el

```diff
@@ -2,21 +2,42 @@
 ;;
 ;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
 ;;
-;; Author: Chris Bowdon <c.bowdon@bath.edu>
-;; URL: https://github.com/syl20bnr/spacemacs
+;; Author: JAremko <w3techplaygound@gmail.com> 
+;; URL: https://github.com/JAremko/spacemacs-pr
 ;;
 ;; This file is not part of GNU Emacs.
 ;;
 ;;; License: GPLv3
 
-(setq typescript-packages '(tss))
+(setq typescript-packages '(tide))
 
-(defun typescript/init-tss ()
-  "Initialize my package"
-  (use-package tss
+(defun typescript/init-tide ()
+  (use-package tide
     :defer t
-    :mode ("\\.ts\\'" . typescript-mode)
-    :init (spacemacs/set-leader-keys-for-major-mode 'typescript-mode
-            "gg" 'tss-jump-to-definition
-            "hh" 'tss-popup-help)
-    :config (tss-config-default)))
+    :init (progn
+            (evilified-state-evilify tide-references-mode tide-references-mode-map
+              (kbd "C-j") 'tide-find-previous-reference
+              (kbd "C-k") 'tide-find-next-reference
+              (kbd "C-l") 'tide-goto-reference)
+            (add-hook 'typescript-mode-hook
+              (lambda ()
+                (tide-setup)
+                (flycheck-mode t)
+                (setq flycheck-check-syntax-automatically '(save mode-enabled))
+                (eldoc-mode t)
+                (when (configuration-layer/package-usedp 'company) 
+                  (company-mode-on)))))
+    :config (progn
+              (spacemacs/declare-prefix-for-mode 'typescript-mode "mg" "goto")
+              (spacemacs/declare-prefix-for-mode 'typescript-mode "mh" "help")
+              (spacemacs/declare-prefix-for-mode 'typescript-mode "mn" "name")
+              (defun typescript-jump-to-type-definition ()
+                (interactive) (tide-jump-to-definition t))
+              (spacemacs/set-leader-keys-for-major-mode 'typescript-mode
+                "gb" 'tide-jump-back
+                "gi" 'typescript-jump-to-type-definition
```

Comment:
Also those key bindings. I couldn't come up with something semantic so I picked `g` `i` because it's somewhat ergonomic :smile: